compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -183,7 +183,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
 
     pub(crate) fn build_reduced_graph(
         &mut self,
-        fragment: &AstFragment,
+        fragment: &mut AstFragment,
```

Comment:
I admit I dislike the mutable def collector a lot.

Maybe make a separate small mut visitor for span marking instead of turning def collector mutable?
Def collector has only 12 calls to `with_parent`, shouldn't be hard to reproduce.